ring-core/src/ring/util/response.clj

```diff
@@ -166,21 +175,36 @@
        (integer? (:status resp))
        (map? (:headers resp))))
 
-(defn- connection-content-length [resp ^java.net.URLConnection conn]
-  (let [content-length (.getContentLength conn)]
-    (if (neg? content-length)
-      resp
-      (header resp "Content-Length" content-length))))
+(defmulti resource-info
+  "Returns information about the resource specified by url.
+
+  The return value is a map with optional values for:
+
+  * :content        - the content of the URL, suitable for use as the :body
```

Comment:
If we follow your idea below about only returning data when the file is not a directory, I think I prefer `resource-file-data`, with the private `file-info` renamed to `file-data`. How does that sound?